flexget/plugins/sites/torrentleech.py

```diff
@@ -121,48 +121,35 @@ def search(self, task, entry, config=None):
             categories = [categories]
         # If there are any text categories, turn them into their id number
         categories = [c if isinstance(c, int) else CATEGORIES[c] for c in categories]
-        filter_url = '/categories/%s' % ','.join(str(c) for c in categories)
+        filter_url = '/categories/{}'.format(','.join(str(c) for c in categories))
         entries = set()
         for search_string in entry.get('search_strings', [entry['title']]):
             query = normalize_unicode(search_string).replace(":", "")
-            # urllib.quote will crash if the unicode string has non ascii characters, so encode in utf-8 beforehand
-            url = ('https://v4.torrentleech.org/torrents/browse/index/query/' +
+            # urllib.quote will crash if the unicode string has non ascii characters,
+            # so encode in utf-8 beforehand
+
+            url = ('https://www.torrentleech.org/torrents/browse/list/query/' +
                    quote(query.encode('utf-8')) + filter_url)
-            log.debug('Using %s as torrentleech search url' % url)
+            log.debug('Using %s as torrentleech search url', url)
 
-            page = task.requests.get(url, cookies=login.cookies).content
-            soup = get_soup(page)
+            results = task.requests.get(url, headers=request_headers, cookies=login.cookies).json()
 
-            for tr in soup.find_all("tr", ["even", "odd"]):
-                # within each even or odd row, find the torrent names
-                link = tr.find("a", attrs={'href': re.compile('/torrent/\d+')})
-                log.debug('link phase: %s' % link.contents[0])
+            for torrent in results['torrentList']:
                 entry = Entry()
-                # extracts the contents of the <a>titlename/<a> tag
-                entry['title'] = link.contents[0]
-
-                # find download link
-                torrent_url = tr.find("a", attrs={'href': re.compile('/download/\d+/.*')}).get('href')
-                # parse link and split along /download/12345 and /name.torrent
-                download_url = re.search('(/download/\d+)/(.+\.torrent)', torrent_url)
-                # change link to rss and splice in rss_key
-                torrent_url = 'https://v4.torrentleech.org/rss' + download_url.group(1) + '/' \
-                              + rss_key + '/' + download_url.group(2)
-                log.debug('RSS-ified download link: %s' % torrent_url)
+                entry['download_headers'] = request_headers
+                entry['title'] = torrent['name']
+
+                # construct download URL
+                torrent_url = 'https://www.torrentleech.org/rss/download/{}/{}/{}'.format(torrent['fid'], rss_key,
+                                                                                          torrent['filename'])
```

Comment:
I don't think this line break is necessary though

Reply:
The limit was 120 chars? Because the line is over that limit.

Reply:
I think my mono-space font is missing.

Reply:
I just use my IDE ruler :)

Reply:
![screenshot 2018-04-11 20 10 57](https://user-images.githubusercontent.com/1884191/38635076-8bd07e84-3dc4-11e8-9856-f68c5f0e5d97.png)